server/pulp/server/db/model/__init__.py

```diff
@@ -51,3 +52,51 @@ class RepositoryContentUnit(Document):
                     'fields': ['unit_id']
                 }
             ]}
+
+
+class Repository(Document):
+    """
+    Defines schema for a pulp repository in the `repos` collection.
+
+    :ivar repo_id: unique across all repos
+    :type repo_id: mongoengine.StringField
+    :ivar display_name: user-readable name of the repository
+    :type display_name: mongoengine.StringField
+    :ivar description: free form text provided by the user to describe the repo
+    :type description: mongoengine.StringField
+    :ivar notes: arbitrary key-value pairs programmatically describing the repo;
+                 these are intended as a way to describe the repo usage or
+                 organizational purposes and should not vary depending on the
+                 actual content of the repo
+    :type notes: mongoengine.DictField
+    :ivar content_unit_counts: key-value pairs of number of units associated with this repo.
+                               This is different than the number of associations, since a
+                               unit may be associated multiple times.
+    :type content_unit_counts: mongoengine.DictField
+    :ivar metadata: arbitrary data that describes the contents of the repo;
+                    the values may change as the contents of the repo change,
+                    either set by the user or by an importer or distributor
+    :type metadata: mongoengine.DictField
+    :ivar _ns: (Deprecated) Namespace of repo, included for backwards compatibility.
+    :type _is: mongoengine.StringField
+    """
+
+    # This field is named `repo_id` because `id`  cannot be accessed using the normal mongoengine
+    # idiom of `obj.id` because `obj.id` has been aliased (internally in mongoengine) to reference
+    # obj._id. So we name this field `repo_id`, and temporarily continue to store it in the db as
+    # `id` for backwards compatibility. This should be migrated to `repo_id` in a future X release.
+    repo_id = StringField(db_field='id', required=True)
+    display_name = StringField(required=True)
```

Comment:
can you explain why have you decided to make display_name as required field and, for example, description not required?